### PR TITLE
Fix Heroku CLI 

### LINKED ISSUE #66: [Bug]: Heroku Download URL not working
### Feature id and version

ghcr.io/devcontainers-extra/features/heroku:1.0.2

### Dev Container base Image

_No response_

### What happened?

- Heroku changed their way of storing versioned CLI paths: https://github.com/heroku/cli/issues/2477, so the current method of string interpolation with the version and machine arch doesn't work anymore. 
- We can also use `channels/stable/` in the download URL for the latest version (https://github.com/devcontainers-contrib/features/pull/575)

https://github.com/devcontainers-extra/features/blob/8a08851a15dcf5774e92c18cfa53bd47ce4e44fb/src/heroku-cli/install.sh#L53-L58

What's nice is that we can use their JSON indices to get the download URL with jq. 

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by joshuanianji**
Fixes #66 

This fix only works for CLI version 8 and above, but I think it's safe to assume people aren't downloading older cli versions. I made a note in the description just in case.

### CODE REVIEW

File: src/heroku-cli/devcontainer-feature.json [hunk 1]
```diff
@@ -2,14 +2,14 @@
     "name": "Heroku CLI (via cli-assets.heroku.com)",
     "id": "heroku-cli",
     "version": "1.0.2",
-    "description": "Heroku CLI allows you to interact with the Heroku API via the command line",
+    "description": "Heroku CLI allows you to interact with the Heroku API via the command line (version 8 and above)",
     "documentationURL": "https://github.com/devcontainers-contrib/features/tree/main/src/heroku-cli",
```
Comment (koralowiec) [inline src/heroku-cli/devcontainer-feature.json:6]: `devcontainers-contrib` -> `devcontainers-extra`

File: test/heroku-cli/scenarios.json [hunk 2]
```diff
@@ -4,5 +4,13 @@
         "features": {
             "heroku-cli": {}
         }
+    },
+    "non_latest_version": {
+        "image": "mcr.microsoft.com/devcontainers/base:debian",
+        "features": {
+            "heroku-cli": {
+                "version": "9.3.2"
```
Comment (koralowiec) [inline test/heroku-cli/scenarios.json:12]: Use a version that's not the latest one, for example `9.3.1`

File: test/heroku-cli/non_latest_version.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -e
+
+source dev-container-features-test-lib
+
+check "heroku existance" heroku version
```
Comment (koralowiec) [inline test/heroku-cli/non_latest_version.sh:7]: Check if the version is correct with:
```bash
check "heroku CLI version is equal to 9.3.1" sh -c "heroku version | grep '9.3.1'"
```

**Review by koralowiec — CHANGES_REQUESTED** [review]
Thanks for the PR! I added some change requests

File: test/heroku-cli/non_latest_version.sh
Comment (koralowiec) [inline test/heroku-cli/non_latest_version.sh:1]: Change file name to `test_non_latest_version.sh`

File: test/heroku-cli/test.sh [hunk 4]
```diff
@@ -5,5 +5,6 @@ set -e
 source dev-container-features-test-lib
 
 check "heroku existance" heroku version
+check "heroku CLI version is equal to 9.3.1" sh -c "heroku version | grep '9.3.1'"
```
Comment (koralowiec) [inline test/heroku-cli/test.sh:8]: Move this line to `test_non_latest_version.sh`, so it's checked for `test_non_latest_version` scenario

**Review by koralowiec — CHANGES_REQUESTED** [review]
Some additional change requests

File: test/heroku-cli/test.sh (hunk 4 above)
Comment (joshuanianji) [reply test/heroku-cli/test.sh:8]: oops lol, that was a dumb mistake

File: src/heroku-cli/devcontainer-feature.json [hunk 5]
```diff
@@ -2,14 +2,14 @@
     "name": "Heroku CLI (via cli-assets.heroku.com)",
     "id": "heroku-cli",
     "version": "1.0.2",
```
Comment (koralowiec) [inline src/heroku-cli/devcontainer-feature.json:4]: Please, bump the version to `1.0.3`. I should have spotted it earlier

**Review by koralowiec — CHANGES_REQUESTED** [review]
Sorry, the last change request :sweat_smile:

File: src/heroku-cli/devcontainer-feature.json (hunk 5 above)
Comment (koralowiec) [reply src/heroku-cli/devcontainer-feature.json:4]: I added #70 to create a check for that in the future

Comment (joshuanianji) [reply src/heroku-cli/devcontainer-feature.json:4]: thanks! good catch

**Review by koralowiec — APPROVED** [review]

**Pull request merged**